iothub/device/samples/how to guides/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -115,40 +121,29 @@ private async Task InitializeAndSetupClientAsync(CancellationToken cancellationT
                             {
                                 await s_deviceClient.CloseAsync(cancellationToken);
                             }
-                            catch (UnauthorizedException) { } // if the previous token is now invalid, this call may fail
+                            catch (UnauthorizedException) { } // If the previous token is now invalid, this call may fail
                             s_deviceClient.Dispose();
                         }
 
-                        s_deviceClient = DeviceClient.CreateFromConnectionString(_deviceConnectionStrings.First(), _transportType, _clientOptions);
+                        s_deviceClient = DeviceClient.CreateFromConnectionString(_deviceConnectionStrings.First(), _transportType, s_clientOptions);
                         s_deviceClient.SetConnectionStatusChangesHandler(ConnectionStatusChangeHandlerAsync);
+                        s_deviceClient.SetRetryPolicy(_customRetryPolicy);
                         _logger.LogDebug("Initialized the client instance.");
+
+                        // Force connection now.
+                        // We have set the "shouldExecuteOperation" function to always try to open the connection.
```

Comment:
This particular statement is no longer required (the "shouldExecuteOperation" logic has been removed)